book/src/pong_tutorial/pong_tutorial_01.md

```diff
@@ -0,0 +1,139 @@
+                    input:
+                        KeyboardInput {
+                            virtual_keycode: Some(VirtualKeyCode::Escape),
+                            ..
+                        },
+                    ..
+                } => Trans::Quit,
+                _ => Trans::None,
+            },
+            _ => Trans::None,
+        }
+    }
+}
+```
+
+The `handle_event` method is executed on every frame before updating, and it's used to react to events. It returns a `Trans`, which is an enum of state machine transitions. In this case, we're watching for the Escape keycode from the Window. If we receive it, we'll return `Trans::Quit` which will be used to clean up the State and close the application. All other keyboard input is ignored for now.
+
+Now that we know we can quit, let's add some code to actually get things started! We'll use a `run()` function, which allows for bettor Error handling:
+
+```rust,ignore
+fn run() -> Result<(), amethyst::Error> {
+
+//We'll put the rest of the code here
+
+  Ok(())
+}
+```
+
+Inside run() we first define a path for our display_config.ron file and load it.
+
+```rust,ignore
+  let path = format!(
+      "{}/resources/display_config.ron",
+      env!("CARGO_MANIFEST_DIR")
+  );
+  let config = DisplayConfig::load(&path);
+```
+
+This .ron file was automatically generated by `amethyst new`. If you didn't use `amethyst new`, now would be a good time to create this config file inside a folder named resources. If you already have this file, we have some changes to make, anyway:
+
+```ron,ignore
+(
+  title: "Pong!",
+  dimensions: Some((500, 500)),
+  max_dimensions: None,
+  min_dimensions: None,
+  fullscreen: false,
+  multisampling: 1,
+  visibility: true,
+  vsync: true,
+)
+```
+
+This will set the default window dimensions to 500 x 500, and make the title bar say "Pong!" instead of the sad, lowercase default of "pong".
+
+Now, back inside our run() function in main.rs, let's copy and paste some rendering code we don't fully understand, just so we can keep moving. The next chapter will cover rendering more in depth.
+
+```rust,ignore
+  let pipe = Pipeline::build().with_stage(
+      Stage::with_backbuffer()
+          .clear_target([0.0, 0.0, 0.0, 1.0], 1.0)
+          .with_pass(DrawFlat::<PosNormTex>::new()),
+  );
+```
+
+The important part of this code is that it renders a black background, because pong is traditionally played on a black background, but you can choose any color you'd like by tweaking the RGBA values inside the `.clear_target` method. Values range from 0.0 to 1.0, so to get that cool green color we started with back, for instance, you can try `[0.00196, 0.23726, 0.21765, 1.0]`.
+
+Now let's pack everything up and run it:
+
+```rust,ignore
+  let mut game = Application::build("./", Pong)?
+      .with_bundle(RenderBundle::new())?
+      .with_local(RenderSystem::build(pipe, Some(config))?)
+      .build()
+      .expect("Fatal error");
```

Comment:
This is probably how we do it in one example, but it's suboptimal. Given that have a `run` function which returns a `Result`, this should just return the error.